containers/daap-create-metadata/src/var/task/create_metadata.py

```diff
@@ -1,74 +1,112 @@
 import json
 import os
+from enum import Enum
 
 from data_platform_logging import DataPlatformLogger
 from data_product_metadata import DataProductMetadata
 
 
+class HTTPStatus(Enum):
+    SUCCESS = 200
+    CREATED = 201
+    BAD_REQUEST = 400
+    METHOD_NOT_ALLOWED = 405
+    CONFLICT = 409
+
+
 def handler(event, context):
+    """
+    Handles requests that are passed through the Amazon API Gateway data-platform/register REST API endpoint.
+    POST requests result in a success code to report that the product has been registered
+    (metadata has been created).
+    GET, PUT, and DELETE requests result in a 405 response.
+
+    Body: 'metadata' must be sent in the request body, encoded as JSON, following the
+          [spec](https://github.com/ministryofjustice/modernisation-platform-environments/blob/main/terraform/environments/data-platform/data-product-metadata-json-schema/v1.0.0/moj_data_product_metadata_spec.json)
+
+    :param event: The event dict sent by Amazon API Gateway that contains all of the
+                  request data.
+    :param context: The context in which the function is called.
+    :return: A response that is sent to Amazon API Gateway, to be wrapped into
+             an HTTP response. The 'statusCode' field is the HTTP status code
+             and the 'body' field is the body of the response.
+    """
+
     logger = DataPlatformLogger(
         extra={
             "image_version": os.getenv("VERSION", "unknown"),
             "base_image_version": os.getenv("BASE_VERSION", "unknown"),
         }
     )
 
+    def format_response(response_code: int, event: dict, body_dict: dict) -> dict:
+        """
+        Generate a response to return to API Gateway that contains the initial event,
+        response code, and contents of the response body
+        (i.e. body['message'] or body['error']['message'])
+        """
+        response_body = {"input": event}
+        response_body.update(body_dict)
+        response = {
```

Comment:
- 1 / 2: I have no preference. And I agree that it may not be useful to the client at the time to include the `input` event. Using the logger in the base image may be the best bet.
- 3: This is a good suggestion. I'll implement it here, then propagate to the base image later.